libraries/chain/include/eosio/chain/database_utils.hpp

```diff
@@ -212,6 +267,34 @@ namespace chainbase {
       fc::raw::unpack(ds, oid._id);
       return ds;
    }
+
+   // chainbase::shared_cow_string
+   // ----------------------------
+   template<typename Stream>
+   inline Stream& operator<<(Stream& s, const chainbase::shared_cow_string& v)  {
+      fc::raw::pack(s, v);
+      return s;
+   }
+
+   template<typename Stream>
+   inline Stream& operator>>(Stream& s, chainbase::shared_cow_string& v)  {
+      fc::raw::unpack(s, v);
+      return s;
+   }
+
+   // chainbase::shared_cow_vector
+   // ----------------------------
+   template<typename Stream, typename T>
+   inline Stream& operator<<(Stream& s, const chainbase::shared_cow_vector<T>& v)  {
+      fc::raw::pack(s, v);
+      return s;
+   }
+
+   template<typename Stream, typename T>
+   inline Stream& operator>>(Stream& s, chainbase::shared_cow_vector<T>& v)  {
+      fc::raw::unpack(s, v);
+      return s;
+   }
```

Comment:
`fc:raw::un/pack()` on a non-reflected class will call `operator<<`/`>>`,
https://github.com/AntelopeIO/leap/blob/f4b7ace8f753a4e2795e342178aed6c0d3114667/libraries/libfc/include/fc/io/raw.hpp#L660-L663
https://github.com/AntelopeIO/leap/blob/f4b7ace8f753a4e2795e342178aed6c0d3114667/libraries/libfc/include/fc/io/raw.hpp#L414-L419
https://github.com/AntelopeIO/leap/blob/f4b7ace8f753a4e2795e342178aed6c0d3114667/libraries/libfc/include/fc/io/raw.hpp#L369-L372

I wonder if we don't need both the un/pack() overrides above and operator<</>> (i.e. the implementations of un/pack() can be moved in to operator<</>>)

Reply:
Updated.